efficientdet/third_party/grad_checkpoint/memory_saving_gradients.py

```diff
@@ -1,21 +1,43 @@
+# Copyright 2020 Google Research. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# This file is a re-distribution based on the copyright in LICENSE.txt
```

Comment:
We are required that every file has this head, but I added a line "This file is a re-distribution based on the copyright in LICENSE.txt" to explain the re-distribution.